### PR TITLE
fix: call destroy on finish to handle node bug

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/ios-deploy.js [hunk 1]
```diff
@@ -45,11 +45,11 @@ class IOSDeploy {
...
       notificationService.observeNotification(APPLICATION_INSTALLED_NOTIFICATION, { notification: resolve });
-    }).timeout(APPLICATION_NOTIFICATION_TIMEOUT);
+    });
     try {
       await installationService.installApplication(bundlePathOnPhone, { PackageType: 'Developer'});
       try {
-        await appInstalledNotification;
+        await appInstalledNotification.timeout(APPLICATION_NOTIFICATION_TIMEOUT);
       } catch (e) {
         log.warn(`Couldn't get the application installed notification with ${APPLICATION_NOTIFICATION_TIMEOUT}ms but we will continue`);
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: how about logging the actual error here?

File: lib/ios-deploy.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ class IOSDeploy {
         } else {
           const readStream = fs.createReadStream(itemPath, {autoClose: true});
           const writeStream = await afcService.createWriteStream(pathOnPhone, {autoDestroy: true });
+          writeStream.on('finish', writeStream.destroy);
```
Comment (mykola-mokhnach) [inline lib/ios-deploy.js]: which argument `writeStream.destroy` is supposed to receive?

Comment (umutuzgur) [reply lib/ios-deploy.js]: Nothing since it ended properly

File: lib/ios-deploy.js (hunk 1 above)
Comment (umutuzgur) [reply lib/ios-deploy.js]: Sure, I can add that

**Pull request merged**